colcon_ros_gradle/package_identification/ros_gradle.py

```diff
@@ -22,134 +22,25 @@ def __init__(self):  # noqa: D107
-        metadata.dependencies['test']  |= data['depends']
-
-def extract_data(build_gradle):
-    """
-    Extract the project name and dependencies from a build.gradle file.
-
-    :param Path build_gradle: The path of the build.gradle file
-    :rtype: dict
-    """
-    # Content for dependencies
-    content_build_gradle = extract_content(build_gradle)
-    match = re.search(r'apply plugin: ("|\')org.ros2.tools.gradle\1', content)
-    if not match:
-        raise RuntimeError("Gradle plugin missing, please add the following to build.gradle: \"apply plugin: 'org.ros2.tools.gradle'\"")
-
-    # Content for name
-    content_setting_gradle = extract_content(build_gradle.parent, 'settings.gradle')
-
-    data = {}
-    data['name'] = extract_project_name(content_setting_gradle)
-    # fallback to the directory name
-    if data['name'] is None:
-        data['name'] = build_gradle.parent.name
-
-    # extract dependencies from all Gradle files in the project directory
-    data['depends'] = set()
+        # Disable build type.
+        desc.type = None
 
-    return data
-
-
-def extract_content(basepath, filename='build.gradle', exclude=None):
-    """
-    Get all non-comment lines from build.gradle files under the given basepath.
-
-    :param Path basepath: The path to recursively crawl
-    :param list exclude: The paths to exclude
-    :rtype: str
-    """
-    if basepath.is_file():
-        content = basepath.read_text(errors='replace')
-    elif basepath.is_dir():
-        content = ''
-        for dirpath, dirnames, filenames in os.walk(str(basepath)):
-            # skip subdirectories starting with a dot
-            dirnames[:] = filter(lambda d: not d.startswith('.'), dirnames)
-            dirnames.sort()
-
-            for name in sorted(filenames):
-                if name != filename:
-                    continue
-
-                path = Path(dirpath) / name
-                if path in (exclude or []):
-                    continue
-
-                content += path.read_text(errors='replace') + '\n'
-    else:
-        return ''
-    return _remove_gradle_comments(content)
-
-
-def _remove_gradle_comments(content):
-    # based on https://stackoverflow.com/questions/241327/python-snippet-to-remove-c-and-c-comments#241506
-    def replacer(match):
-        s = match.group(0)
-        if s.startswith('/'):
-            return " " # note: a space and not an empty string
-        else:
-            return s
-    pattern = re.compile(
-        r'//.*?$|/\*.*?\*/|\'(?:\\.|[^\\\'])*\'|"(?:\\.|[^\\"])*"',
-        re.DOTALL | re.MULTILINE
-    )
-    return re.sub(pattern, replacer, content)
-
-def extract_project_name(content):
-    """
-    Extract the Gradle project name from the Gradle settings file.
+        # Call Gradle package identification extention.
+        gradle_extention = GradlePackageIdentification()
+        gradle_extention.identify(desc)
+        if desc.type != 'gradle':
+            return
```

Comment:
This justifies some kind of warning imo. The package manifest declares to be of build type `gradle` but the later check indicates it isn't actually one...